[Onechat] Small UI polish tasks for one chat

chrisbmar opened the pull request:
## Summary

### Description
- [x] Conversation history pane is no longer collapsed by default, it defaults to being **open**
- [x] Updated the default agent name from `onechat default agent` to `Default Agent`
- [x] Fixed a bug with how we use react-query in `use_conversation_list.ts`, we were relying on using `isLoading` to conditionally render different things in the UI, however, `isLoading` will always be `false` when we provide `initialData` to the query.
- [x] Default Agent Edit link: remove the edit link to default agent.
https://github.com/user-attachments/assets/a99b3c74-3bbd-4533-b5a5-ef583d9165bf
- [x] Flash for new conversation: fix this converation flash

chrisbmar committed:
set conversation history pane open by default

chrisbmar committed:
rename default agent definition name and description

chrisbmar committed:
remove initial data from use conversation list query as it stops isLoading ever being true

chrisbmar committed:
provide default value for useConversationList in conversations_view

chrisbmar committed:
add util and test for getting the most recent conversation

chrisbmar committed:
when navigating back to chat, open the most recent conversation

chrisbmar commented on x-pack/platform/plugins/shared/onechat/public/application/pages/conversations.tsx:
```diff
@@ -5,18 +5,46 @@
+  }, [conversationId, conversations, isLoading, navigateToOnechatUrl]);
+
+  // Redirecting /conversations to most recent conversation or /new
+  if (!conversationId) {
+    return null;
+  }
```
Alternatively could show a spinner here if we're concerned about the `api/chat/conversations` query potentially taking a while to resolve. I checked ChatGPT for reference and they don't show any loading spinners whilst loading conversations.

elasticmachine committed:
Merge branch 'main' into onechat-10847-ui-polish-1

chrisbmar committed:
remove clone option for default agent and remove link to edit agent

elasticmachine committed:
Merge branch 'main' into onechat-10847-ui-polish-1

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/335912)
* Commit: db4a620ba199a92323d1e0efa2d7d3f80fe787cd

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/d91200b5535a82c3987afe017462b02c27c13c7c) | [after](https://github.com/elastic/kibana/commit/db4a620ba199a92323d1e0efa2d7d3f80fe787cd) | diff |
| --- | --- | --- | --- |
| `onechat` | 478 | 479 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d91200b5535a82c3987afe017462b02c27c13c7c) | [after](https://github.com/elastic/kibana/commit/db4a620ba199a92323d1e0efa2d7d3f80fe787cd) | diff |
| --- | --- | --- | --- |
| `onechat` | 829.0KB | 829.4KB | +416.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/d91200b5535a82c3987afe017462b02c27c13c7c) | [after](https://github.com/elastic/kibana/commit/db4a620ba199a92323d1e0efa2d7d3f80fe787cd) | diff |
| --- | --- | --- | --- |
| `onechat` | 12.9KB | 12.9KB | +1.0B |



<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335912","number":335912,"commit":"db4a620ba199a92323d1e0efa2d7d3f80fe787cd"}],"number":335912}
buildkite-pr-comment-->

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/public/application/utils/get_most_recent_conversation.test.ts:
Thanks for adding a test!

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/public/application/pages/conversations.tsx:
```diff
@@ -5,18 +5,46 @@
     },
   ]);
+
+  useEffect(() => {
```
Yeah fair, it isn't totally necessary. But in general I put `useEffect` hooks into custom hooks with names that describe what they're doing. It's really just for documentation. This isn't really true for this component because it is really small, but if you create a custom hook for the logic then it is clear what the dependencies you added are being used for. 

For example I recently created `useSyncAgentId` https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/onechat/public/application/hooks/use_sync_agent_id.ts
This is only ever going to be used in one place, but I think it makes it clearer what the effect's purpose is.

That hook is used in `Conversation`, if I put the effect inline in that component with all of its dependencies you'd have to read the code to understand why `Conversation` needs access to each of those dependencies. And then you'd also need a comment above the effect to describe what it is doing. I think it's better to abstract the logic so that the component can stay leaner.

zacharyparikh replied:
Yeah there's definitely pros/cons with both approaches. How you have it is good.

chrisbmar committed:
revert showing most recent conversation first

chrisbmar committed:
remove line

chrisbmar committed:
bring back agent clone option

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/336167)
* Commit: 78efb7db6af38e5385d6400940c98a09c5406f8f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d91200b5535a82c3987afe017462b02c27c13c7c) | [after](https://github.com/elastic/kibana/commit/78efb7db6af38e5385d6400940c98a09c5406f8f) | diff |
| --- | --- | --- | --- |
| `onechat` | 829.0KB | 829.0KB | +29.0B |



### History
* :green_heart: [Build #335912](https://buildkite.com/elastic/kibana-pull-request/builds/335912) succeeded db4a620ba199a92323d1e0efa2d7d3f80fe787cd

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/336167","number":336167,"commit":"78efb7db6af38e5385d6400940c98a09c5406f8f"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/335912","number":335912,"commit":"db4a620ba199a92323d1e0efa2d7d3f80fe787cd"}],"number":336167}
buildkite-pr-comment-->

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

chrisbmar committed:
agent and tool naming capitalisation

chrisbmar committed:
stop flashing new conversation

chrisbmar commented:
@zacharyparikh let me know what you think about [this](https://github.com/elastic/kibana/pull/233964/commits/1f871f574a4ce7c4b7e7220c90a286f90a58f8f4) commit - I added it to stop the flashing of "_New conversation_" in the the title of existing conversations. 

I think we need to consider the loading states of queries in a more robust way to avoid flashing when there isn't data...and then there is. The solution I added now is to fit with the existing structure of having hooks which 'describe' their functionality I.e. `useConversationTitle`. This approach assumes that we can get data in a synchronous way, which we can't do when making an async request, it also means we need to pass the loading state through all the hooks if the UI needs to know about it.

chrisbmar commented on x-pack/platform/plugins/shared/onechat/public/application/components/agents/list/agents.tsx:
```diff
@@ -31,7 +31,7 @@ export const OnechatAgents = () => {
     >
       {i18n.translate('xpack.onechat.agents.newAgentButton', {
-        defaultMessage: 'New agent',
+        defaultMessage: 'New Agent',
```
This was a specific ask by @joemcelroy on a call earlier, he updated the spec in the epic ticket. 🤷🏽

zacharyparikh replied:
If the goal is just to have button consistency, then I think we should standardize on sentence casing

julianrosado commented:
Before committing UX copy changes, let just get the blessing of @leemthompo !

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_title.tsx:
```diff
@@ -35,9 +35,11 @@ export const ConversationTitle: React.FC<{}> = () => {
-      </EuiTitle>
+      {!isLoading && (
+        <EuiTitle aria-label={labels.ariaLabel} size="xxs">
+          <h1>{title || labels.newConversationDisplay}</h1>
```
Ah right, of course. Yeah I think this is the right approach then, I think it's better to show nothing for a few ms than flickering a skeleton.

zacharyparikh replied:
I don't actually think a skeleton is really a good idea here, because the loading state is so short lived. I think the blank state isn't very noticeable.

julianrosado replied:
Fair, unless we artificially increase the loading state to make it a smoother transition 🤔🤔🤔 No Julian, you're mad my precious. Ok sssh, they're listening my precious. 

All good with going with this for M2! 🤗

chrisbmar replied:
I agree with Zachary. I think we should ship as is.

I think we should discuss how we fetch data in the app and expose this to UI components in a more async-friendly way, and build appropriate loading states around this.

leemthompo commented:
I agree with @zacharyparikh about the sentence casing, we should follow the EUI guidelines for copy and be consistent

chrisbmar committed:
fix sentence casing

chrisbmar commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into onechat-10847-ui-polish-1
